src/main/java/org/openjdk/shenandoah/ToolbarPanel.java

```diff
@@ -122,6 +122,7 @@ public ToolbarPanel(boolean isReplay) {
 
         fileNameField = new JTextField();
         fileNameField.setEditable(false);
+        fileNameField.setFocusable(false);
```

Comment:
Is this the fix for the deadlock issue?

Reply:
Yes by moving the `setFoucusable` method, it fixes the deadlock issue.

Reply:
Hmm... I don't really think that's enough. It's an intermittent issue and it may not have deadlocked the last time it was tested, but the root cause is having a non-gui thread (which calls the `RenderPlayback::run` method) manipulate gui components like the slider (`ToolbarPanel::setValue`). That method needs to run on the event queue thread. To be safe, the `repaint` method should also run on the event queue thread (but I think the implementation of that is thread safe).

Reply:
The main cause of the deadlock is that I originally add `setFocusable` method in the `ToolbarPanel::setValue` and I think without this method it won't lock when the `run` method called the `setValue` method

Reply:
So this deadlock situation won't happen

Reply:
However, I still change the `setValue` method so now it runs on the event queue thread.

Reply:
Thank you.